collector/agent_poller.go

```diff
@@ -383,12 +415,12 @@ func getAgentState(agentIp string, stats chan<- StatsEvent) (*AgentState, error)
 		executorAppNames: executorAppNames}, nil
 }
 
-func getJsonFromAgent(agentIp string, urlPath string, testFileFlag *string, stats chan<- StatsEvent) (*jason.Value, error) {
+func (a *Agent) getJsonFromAgent(urlPath string, testFileFlag *string, stats chan<- StatsEvent) (*jason.Value, error) {
 	stats <- MakeEvent(AgentQuery)
 	var rawJson []byte = nil
 	var err error = nil
 	if len(*testFileFlag) == 0 {
-		endpoint := fmt.Sprintf("http://%s:%d%s", agentIp, *agentPortFlag, urlPath)
+		endpoint := fmt.Sprintf("http://%s:%d%s", a.AgentIP, a.Port, urlPath)
```

Comment:
Today, these test flags are scattered around the library. I did not want to remove them in this PR since i'm not 100% sure how to use them properly. This PR removes the main configuration flags. In a later iteration I want to move the test flags into the executable code configuration, and leverage URL to validate the flag options there.